### ISSUE TITLE
Do not call sys.exc_clear() on Python 3 in bin/trial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8467]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8467|
|Type|enhancement|
|Created|2016-06-13 21:59:52Z|
|Branch|https://github.com/twisted/twisted/tree/8467-rodrigc-exc_clear|

sys.exc_clear() is gone on Python 3: https://docs.python.org/3/whatsnew/3.0.html#index-22

We should not try to call it on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__8467 8467
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8467_rodrigc_exc_clear 8467-rodrigc-exc_clear
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__core core
keywords__None None
time__1465855192008856 1465855192008856
changetime__1471513868531850 1471513868531850
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/186

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Here's some review that applies not only to this but to [#8465](https://github.com/twisted/twisted/issues/8465), [#8466](https://github.com/twisted/twisted/issues/8466), [#8468](https://github.com/twisted/twisted/issues/8468), and [#8469](https://github.com/twisted/twisted/issues/8469): what the hell is `exc_clear()` even doing there? It seems to serve no purpose whatsoever. Glyph has suggested it has something to do with a Python 2.2 implementation detail, so I'm inclined to say this is crazy-talk.

I'm going to let this build run, but I'd like to see one of two things:

- Either more aggressive patch that simply changes the except block to `pass`.
- Or the most aggressive patch, which removes all of this nonsense and makes these all `console_scripts`, such that `pip install -e '.[dev]'` (the command we tell people to run for development anyway) makes those scripts available directly.

I'll happily accept either kind of patch, but we should definitely do one of them.

I'll apply this feedback to the other tickets as well. Feel free to ping me directly when you're ready for more review.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @Lukasa|
|-|-|

I updated the branch: https://github.com/twisted/twisted/compare/trunk...8467-rodrigc-exc_clear

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @rodrigc|
|-|-|

Going in a different direction on this after talking to Lukasa

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc commented|
|-|-|

Putting back into review pile:
https://github.com/twisted/twisted/pull/338

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Thanks for the cleanup, rodrigc!

I'm not entirely clear on why `bin/trial` even still exists, since we _have_ already made it a `console_script` entry point.  But the need for `exc_clear` is definitely long past (I do remember, and could explain, but it's irrelevant archaeology at this point) and the more lines we can remove from this the closer we are to eliminating it entirely :).

Just waiting on a successful CI run to merge; if I get distracted and someone else notices a passing build before I get back to it, please land.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_7"></a>|@rodrigc commented|
|-|-|

bin/trial still exists because some of the buildbots still do
"python bin/trial".

My preference is to delete bin/trial from the tree completely,
and have the remaining buildbots change to invoke trial from inside tox,
so that we have consistency between Appveyor, Travis, and Buildbot.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc set status to `closed`|
|-|-|

It's better to delete bin/trial from the source tree
